cgmes/cgmes-model/src/main/java/com/powsybl/cgmes/model/CgmesNamespace.java

```diff
@@ -54,85 +49,239 @@ private CgmesNamespace() {
     public static final String CIM_100_SV_PROFILE = "http://iec.ch/TC57/ns/CIM/StateVariables-EU/3.0";
     public static final String CIM_100_SSH_PROFILE = "http://iec.ch/TC57/ns/CIM/SteadyStateHypothesis-EU/3.0";
 
-    private static final Map<Integer, Map<String, String>> PROFILES = Map.of(
-            16, Map.of("EQ", CIM_16_EQ_PROFILE, "EQ_OP", CIM_16_EQ_OPERATION_PROFILE, "SSH", CIM_16_SSH_PROFILE, "SV", CIM_16_SV_PROFILE, "TP", CIM_16_TP_PROFILE),
-            100, Map.of("EQ", CIM_100_EQ_PROFILE, "EQ_OP", CIM_100_EQ_OPERATION_PROFILE, "SSH", CIM_100_SSH_PROFILE, "SV", CIM_100_SV_PROFILE, "TP", CIM_100_TP_PROFILE)
-    );
+    public static final Cim CIM_14 = new Cim14();
+    public static final Cim CIM_16 = new Cim16();
+    public static final Cim CIM_100 = new Cim100();
 
     public static boolean isValid(String ns) {
         // Until CIM16 the CIM namespace contained the string "CIM-schema-cim<versionNumber>#"
         // Since CIM100 the namespace seems to follow the pattern "/CIM<versionNumber>#"
         return VALID_CIM_NAMESPACES.contains(ns) || CIM_100_PLUS_NAMESPACE_PATTERN.matcher(ns).matches();
     }
 
-    public static String getCim(int cimVersion) {
-        if (CIM_NAMESPACES.containsKey(cimVersion)) {
-            return CIM_NAMESPACES.get(cimVersion);
-        }
-        throw new AssertionError("Unsupported CIM version " + cimVersion);
+    public interface Cim {
+        int getVersion();
+
+        String getNamespace();
+
+        boolean hasProfiles();
+
+        String getProfile(String profile);
+
+        String getEuPrefix();
+
+        String getEuNamespace();
+
+        String getLimitValueAttributeName();
+
+        String getLimitTypeAttributeName();
+
+        String getLimitKindClassName();
+
+        boolean writeLimitInfiniteDuration();
+
+        boolean writeGeneratingUnitInitialP();
     }
 
-    public static String getEu(int cimVersion) {
-        if (cimVersion == 16) {
-            return ENTSOE_NAMESPACE;
-        } else if (cimVersion >= 100) {
-            return EU_NAMESPACE;
+    private static final class Cim14 implements Cim {
+        @Override
+        public int getVersion() {
+            return 14;
+        }
+
+        @Override
+        public String getNamespace() {
+            return CIM_14_NAMESPACE;
+        }
+
+        @Override
+        public boolean hasProfiles() {
+            return false;
+        }
+
+        @Override
+        public String getProfile(String profile) {
+            throw new AssertionError("Unsupported CIM version 14");
+        }
+
+        @Override
+        public String getEuPrefix() {
+            throw new PowsyblException("Undefined eu prefix for version 14");
```

Comment:
nice refactoring!
beware of error messages in Cim14: they all use the same description

Reply:
You're right!